ui/packages/catalog-ui-search/src/main/webapp/component/search-form-editor/search-form-editor.view.js

```diff
@@ -0,0 +1,134 @@
+const SelectionInterface = require('component/selection-interface/selection-interface.model')
+const QueryAdd = require('component/query-add/query-add.view')
+const QueryAdvanced = require('component/query-advanced/query-advanced.view')
+const QueryModel = require('js/model/Query')
+const QueryTitle = require('component/query-title/query-title.view')
+const LoadingView = require('component/loading/loading.view')
+const wreqr = require('wreqr')
+const user = require('component/singletons/user-instance')
+const cql = require('js/cql')
+const announcement = require('component/announcement')
+const $ = require('jquery')
+
+module.exports = Marionette.LayoutView.extend({
+  tagName: CustomElements.register('search-form-editor'),
+  template: template,
+  regions: {
+    queryTitle: '.search-form-editor-title',
+    editor: '.search-form-editor-inputs',
+    map: '.search-form-editor-map',
+  },
+  initialize() {
+    this.listenTo(Router, 'change', this.render)
+    const [id] = Router.get('args')
+  },
+  onRender() {
+    const [id] = Router.get('args')
+    const queryModel =
+      id === 'create'
+        ? new QueryModel.Model()
+        : SearchFormsCollection.getCollection().get(id)
+    this.model = queryModel
+    this.map.show(
+      new MapView({
+        selectionInterface: new SelectionInterface(),
+      })
+    ),
+      this.editor.show(
+        new QueryAdvanced({
+          model: this.model,
+          isForm: true,
+          isFormBuilder: true,
+          onSave: () => {
+            this.saveTemplateToBackend()
+          },
+        })
+      )
+    this.queryTitle.show(
+      new QueryTitle({
+        model: this.model,
+      })
+    )
+  },
+  getQueryAsQueryTemplate: function() {
+    const formModel =
+      this.model.get('associatedFormModel') || new SearchFormModel()
+    const formParameters = this.editor.currentView.serializeTemplateParameters()
+    let filterTree = cql.simplify(formParameters.filterTree || {})
+    let filterSettings = formParameters.filterSettings || {}
+    if (filterTree.filters && filterTree.filters.length === 1) {
+      filterTree = filterTree.filters[0]
+    }
+    filterSettings.sorts = filterSettings.sorts
+      .filter(sort => sort.attribute && sort.direction)
+      .map(sort => sort.attribute + ',' + sort.direction)
+    return {
+      filterTemplate: filterTree,
+      accessIndividuals: formModel.get('accessIndividuals'),
+      accessGroups: formModel.get('accessGroups'),
+      accessAdministrators: formModel.get('accessAdministrators'),
+      creator: formModel.get('createdBy'),
+      id: formModel.get('id'),
+      title: this.model.get('title'),
+      description: formModel.get('description'),
+      created: formModel.get('createdOn'),
+      owner: formModel.get('owner'),
+      querySettings: filterSettings,
+    }
+  },
+  saveTemplateToBackend: function() {
+    let loadingView = new LoadingView()
+    let _this = this
+    let _user = user
```

Comment:
❓ What's the purpose of these two variables? Why not just use `this` and `user`?